python/feathub/processors/spark/dataframe_builder/join_utils.py

```diff
@@ -0,0 +1,145 @@
+#  Copyright 2022 The Feathub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Sequence, Optional
+
+from feathub.common.exceptions import FeathubException
+from feathub.feature_views.sliding_feature_view import (
+    SlidingFeatureView,
+    ENABLE_EMPTY_WINDOW_OUTPUT_CONFIG,
+)
+from feathub.feature_views.transforms.sliding_window_transform import (
+    SlidingWindowTransform,
+)
+from feathub.processors.constants import EVENT_TIME_ATTRIBUTE_NAME
+from feathub.processors.spark.spark_types_utils import to_spark_type
+from feathub.table.table_descriptor import TableDescriptor
+from pyspark.sql import DataFrame as NativeSparkDataFrame, SparkSession
+from pyspark.sql.types import DataType
+
+
+class JoinFieldDescriptor:
+    """
+    Descriptor of the join field.
+    """
+
+    def __init__(
+        self,
+        field_name: str,
+        field_data_type: Optional[DataType] = None,
+    ):
+        """
+        :param field_name: The name of the field to join.
+        :param field_data_type: Optional. If it is not None, the field is cast to the
+                                given type. Otherwise, use its original type.
+        """
+        self.field_name = field_name
+        self.field_data_type = field_data_type
+
+    @staticmethod
+    def from_table_descriptor_and_field_name(
+        table_descriptor: TableDescriptor, field_name: str
+    ) -> "JoinFieldDescriptor":
+        feature = table_descriptor.get_feature(field_name)
+        transform = feature.transform
+
+        if (
+            not isinstance(table_descriptor, SlidingFeatureView)
```

Comment:
SlidingFeatureView is currently not supported in the Spark processor yet. Let's avoid making this complicated. Same for the `SlidingWindowTransform` and `ENABLE_EMPTY_WINDOW_OUTPUT_CONFIG`.